utils/mgmt_system/openstack.py

```diff
@@ -201,7 +201,16 @@ def create_vm(self):
     def delete_vm(self, instance_name):
         logger.info(" Deleting OpenStack instance %s" % instance_name)
         instance = self._find_instance_by_name(instance_name)
+        ip_addr = self.current_ip_address(instance_name)
+        floating_ip = None
+        if ip_addr is not None:
+            floating_ips = self.api.floating_ips.findall(ip=ip_addr)
+            if floating_ips:
+                floating_ip = floating_ips[0]
         instance.delete()
+        if floating_ip is not None:
+            logger.info("Deleting floating IP {}".format(floating_ip.ip))
```

Comment:
It is not a bug, what I posted is a RFE. Deleting the FIP is safer than leaving it in the pool because it is the pool where things can get some race. This enables us to deactivate the floating IP killer which would make it difficult to test openstack machines with publicly available IPs so the deleting must occur. The FIP reusal is just a beneficial side effect.